packages/walletconnect/src/index.ts

```diff
@@ -124,17 +124,14 @@ export class WalletConnect extends Connector {
     const cancelActivation = this.actions.startActivation()
 
     try {
-      await this.isomorphicInitialize()
-      if (!this.provider?.connected) throw Error('No existing connection')
-
-      // Wallets may resolve eth_chainId and hang on eth_accounts pending user interaction, which may include changing
-      // chains; they should be requested serially, with accounts first, so that the chainId can settle.
-      const accounts = await this.provider.request<string[]>({ method: 'eth_accounts' })
-      if (!accounts.length) throw new Error('No accounts returned')
-      const chainId = await this.provider.request<string>({ method: 'eth_chainId' })
-
-      this.actions.update({ chainId: parseChainId(chainId), accounts })
+      const { session, accounts, chainId } = await this.isomorphicInitialize()
```

Comment:
nit: for consistency with `activate`, can you assign to `provider` and then access properties via provider?

```suggestion
      const provider = await this.isomorphicInitialize()
```

Reply:
Personally, don't have preference here, so I am happy to take this.